astropy/cosmology/tests/test_flrw.py

```diff
@@ -12,15 +12,33 @@
 import astropy.units as u
 from astropy.cosmology import (FLRW, FlatLambdaCDM, Flatw0waCDM, FlatwCDM,
                                LambdaCDM, Planck18, w0waCDM, w0wzCDM, wCDM, wpwaCDM)
+from astropy.cosmology.flrw import ellipkinc, hyp2f1, quad
+from astropy.utils.compat.optional_deps import HAS_SCIPY
 
-from .test_core import FlatCosmologyMixinTest
 from .test_core import CosmologySubclassTest as CosmologyTest
+from .test_core import FlatCosmologyMixinTest
 
 ##############################################################################
 # TESTS
 ##############################################################################
 
 
+@pytest.mark.skipif(HAS_SCIPY, reason="can't have :mod:`scipy` imported")
```

Comment:
No need to be too fancy with the reason here. It doesn't even show up in the test log and definitely will not be rendered by anything user facing. It is just a note for devs.
```suggestion
@pytest.mark.skipif(HAS_SCIPY, reason="scipy is missing")
```

Reply:
Yes, just at very first reading it did not make much sense to me.

Reply:
SGTM.

Reply:
On a related note, this test doesn't seem to be running! I don't see any reason why it doesn't run (except when scipy is installed)!

Reply:
You mean, it doesn't run in CI or locally? Check the pytest header. Maybe somehow all our jobs pull in scipy.

Reply:
The 'minimal dependencies' ones of course don't, but I thought I had in fact seen everything in `test_flrw.py` run in the previous run.

Reply:
Well... hopefully not but you can never be sure that another thingy we're pulling into the job isn't pulling in scipy. There was just discussions on how scipy isn't really optional but I forgot the context.

Reply:
The first one is skipped on '3.8 with oldest dependencies' (with Scipy 1.3.3):
```
../../.tox/py38-test-oldestdeps/lib/python3.8/site-packages/astropy/cosmology/tests/test_flrw.py s [ 18%]
296
......................................................                   [ 18%]
297
```
but not on '3.9 with minimal dependencies' (no Scipy)
```
../../.tox/py39-test/lib/python3.9/site-packages/astropy/cosmology/tests/test_flrw.py . [ 18%]
271
......................................................                   [ 19%]
```
– looks as it should be.

Reply:
@dhomeier Thanks for the diagnostic!
But somehow codecov is saying it still isn't tested!

Reply:
I have no idea how codecov determined that. I am seeing those functions marked (mostly) red in https://codecov.io/gh/astropy/astropy/compare/d1313bbb314ec90ea7c16dfac3be6586af1402ae...269d99f979fd1b1dd4ab5e2be6962a269f149781/src/astropy/cosmology/flrw.py#
but no further explanation; maybe it's because of the mark.skipif()?

Reply:
Maybe? I'm pretty mystified. There's nothing that should skip the whole file if scipy isn't present... unless ``__doctest_requires__ = {'*': ['scipy']}`` is the culprit ‽‽

Reply:
Ah, that's because coverage is only calculate for one of the jobs... 😅

Reply:
Ohh! We should fix that. Codecov can merge the coverage.
Followup PR?

Reply:
Add coverage to Python 3.9 with minimal dependencies?

Reply:
change it to

```
          - name: Python 3.9 with minimal dependencies and full coverage
            os: ubuntu-latest
            python: 3.9
            toxenv: py39-test-cov-clocale
```

Reply:
Yes, let's explore that as a separate PR. 😸

Reply:
Yes, testing both minimal and full dependencies sounds sensible.
Wondering how is this going to be evaluated, e.g. if one job reports increased and the other decreased coverage... 👻